includes/snmp.inc.php

```diff
@@ -305,6 +305,10 @@ function snmp_get($device, $oid, $options = null, $mib = null, $mibdir = null)
     $measure->manager()->recordSnmp($measure->end());
     if (preg_match('/(No Such Instance|No Such Object|No more variables left|Authentication failure)/i', $data)) {
         return false;
+    } elseif (preg_match('/Wrong Type(.*)should be/', $data)) {
+        $data = preg_replace('/Wrong Type \(should be .*\): /', '', $data);
```

Comment:
Super nitpick is to rewrite it in a slightly more efficient way, `/Wrong Type \(should be [^:]+: /`

Reply:
I just copied https://github.com/librenms/librenms/blob/master/includes/snmp.inc.php#L424

Reply:
fwiw we have a few instances of 'handling' this

We have
 snmp_get
   https://github.com/librenms/librenms/blob/master/includes/snmp.inc.php#L302
 snmp_walk
   https://github.com/librenms/librenms/blob/master/includes/snmp.inc.php#L424
 snmp_get_multi_oid
   https://github.com/librenms/librenms/blob/master/includes/snmp.inc.php#L252

 tests/Unit/SnmpResponseTest.php
 testErrorHandling
   https://github.com/librenms/librenms/blob/master/tests/Unit/SnmpResponseTest.php#L280

 LibreNMS/Util/ModuleTestHelper.php
 convertSnmpToSnmprec
   https://github.com/librenms/librenms/blob/master/LibreNMS/Util/ModuleTestHelper.php#L374

 LibreNMS/Data/Source/SnmpResponse.php
   __construct
   https://github.com/librenms/librenms/blob/master/LibreNMS/Data/Source/SnmpResponse.php#L55

🤷‍♂️

Help?

Reply:
Just convert everything to SnmpQuery :D